tools/secure_notebook.py

```diff
@@ -66,16 +78,33 @@ def create_self_signed_cert(cert_dir, keyfile, certfiile):
     except ConfigFileNotFound:
         config = Config()
 
-    config.NotebookApp.password = pw
+    yield config
 
     with io.open(os.path.join(jupyter_config_dir(), 'jupyter_notebook_config.json'), 'w') as f:
```

Comment:
I think though that standard practice for these kinds of files is to still store them with tight permissions, even if they are in a 700 dir, in case they get temporarily/accidentally moved out.